ocaml/xapi/quicktest_cbt.ml

```diff
@@ -47,61 +54,106 @@ let make_vdi_from ~session_id ~sR = (* SR has VDI.create as allowed *)
     ~sm_config:[]
 
 
-(**** Test declarations ****)
+(* ******************
+ * Test declarations
+ * ******************)
 
-(* Test enable/disable CBT, test cbt_enabled:false for new VDI *)
-let enable_disable_cbt_test ~session_id ~vDI =
-  let enable_cbt_test = make_test "Testing VDI.enable/disable_CBT" 4 in
+(* Note that tests including expensive VDI operations (snapshot, clone, copy etc)
+ * output debug info at most steps to justify waiting time to user *)
+
+(* Test enable/disable_cbt, data_destroy, and snapshot update the necessary fields *)
+let vdi_data_destroy_test ~session_id ~vDI =
+  let test = make_test "Testing VDI.{enable/disable_cbt, data_destroy, snapshot}" 4 in
   try
-    start enable_cbt_test;
-    test_assert ~test:enable_cbt_test
-      (not (get_cbt_status ~session_id ~vDI))
-      ~msg:"VDI.cbt_enabled field should be set to false for new VDIs";
+    start test;
+    let debug_test msg = debug test msg in
+    debug_test "Enabling CBT on original VDI";
     VDI.enable_cbt ~session_id ~rpc:!rpc ~self:vDI;
-    test_assert ~test:enable_cbt_test
+    test_assert ~test
       (get_cbt_status ~session_id ~vDI)
       ~msg:"VDI.enable_cbt failed";
+
+    debug_test "Snapshotting original VDI with CBT enabled";
+    let snapshot = VDI.snapshot ~session_id ~rpc:!rpc ~vdi:vDI ~driver_params:[] in
+    test_assert ~test
+      (get_cbt_status ~session_id ~vDI:snapshot)
+      ~msg:"VDI.snapshot failed, cbt_enabled field didn't carry over";
+
+    debug_test "Disabling CBT on original VDI";
     VDI.disable_cbt ~session_id ~rpc:!rpc ~self:vDI;
-    test_assert ~test:enable_cbt_test
-      (not (get_cbt_status ~session_id ~vDI)) (* disable_cbt fails *)
-      ~msg:"VDI.disable_CBT failed";
-    success enable_cbt_test
+    test_assert ~test
```

Comment:
We could do the same thing for `test_assert` as for `debug`, to simplify all these assertions:
```
let debug_test msg = debug test msg in
let test_assert = test_assert ~test in
...
```
and then we won't have to pass the `~test` param to `test_assert`.